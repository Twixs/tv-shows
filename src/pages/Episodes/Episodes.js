import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getEpisodeDataStart, selectEpisodes } from '../../store/slices';
import { InfoBlock } from '../../components/InfoBlock/InfoBlock';
import s from './Episodes.module.scss';

export const Episodes = () => {
  const { episodeId } = useParams();
  const dispatch = useDispatch();
  const { current, loading } = useSelector(selectEpisodes);

  useEffect(() => {
    dispatch(getEpisodeDataStart(episodeId));
  }, [dispatch, episodeId]);

  if (loading || !current) {
    return <p>Loading...</p>;
  }

  return (
    <div className={s.EpisodeWrapper}>
      <InfoBlock data={current} />
    </div>
  );
};
