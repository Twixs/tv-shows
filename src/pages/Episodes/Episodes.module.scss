.EpisodeWrapper {
  width: 80%;
  margin: 0 auto;
}