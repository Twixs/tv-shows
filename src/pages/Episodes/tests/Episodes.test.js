import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { getByText, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Episodes } from '../Episodes';

const loadingState = {
  loading: true,
  current: null,
  error: false,
};

const currentItemState = {
  loading: false,
  current: {
    id: 1,
    name: 'My test',
    summary: '<p>Testing test</p>',
  },
  error: false,
};

const mockStore = configureMockStore();

describe('Episodes Page', () => {
  const history = createMemoryHistory();
  history.push('/episodes/1');

  it('has loading message', () => {
    const store = mockStore({ episodes: loadingState });
    const { getByText } = render(
      <Router history={history}>
        <Provider store={store}>
          <Episodes />
        </Provider>
      </Router>
    );

    expect(getByText(/loading\.\.\./i)).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const store = mockStore({ episodes: currentItemState });
    const tree = renderer
      .create(
        <Router history={history}>
          <Provider store={store}>
            <Episodes />
          </Provider>
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
