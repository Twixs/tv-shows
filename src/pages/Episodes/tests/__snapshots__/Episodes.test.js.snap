// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episodes Page renders correctly 1`] = `
<div
  className="EpisodeWrapper"
>
  <div
    className="InfoWrapper"
  >
    <img
      alt="My test"
    />
    <div
      className="Details"
    >
      <h2>
        My test
      </h2>
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>Testing test</p>",
          }
        }
      />
    </div>
  </div>
</div>
`;
